apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,734 @@
+    }
+
+    /**
+    * @dev Return the caster of a given vote by its ID
+    * @param _voteId Vote identifier
+    * @param _voter Address of the voter
+    * @return Address of the caster of the voter's vote
+    */
+    function getVoteCaster(uint256 _voteId, address _voter) external view voteExists(_voteId) returns (address) {
+        return _voteCaster(votes[_voteId], _voter);
+    }
+
+    /**
+    * @notice Tells whether `_representative` is `_voter`'s representative or not
+    * @return True if the given representative was allowed by a certain voter, false otherwise
+    */
+    function isRepresentativeOf(address _voter, address _representative) external view isInitialized returns (bool) {
+        return _isRepresentativeOf(_voter, _representative);
+    }
+
+    /**
+    * @notice Tells whether vote #`_voteId` is within the overrule period for delegated votes or not
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @return True if the requested vote is within the overrule window for delegated votes, false otherwise
+    */
+    function withinOverruleWindow(uint256 _voteId) external view voteExists(_voteId) returns (bool) {
+        Vote storage vote_ = votes[_voteId];
+        return _isVoteOpen(vote_) && _withinOverruleWindow(vote_);
+    }
+
+    // Forwarding fns
+
+    /**
+    * @notice Creates a vote to execute the desired action, and casts a support vote if possible
+    * @dev IForwarder interface conformance
+    *      Disputable apps are required to be the initial step in the forwarding chain
+    * @param _evmScript EVM script to be executed on approval
+    */
+    function forward(bytes _evmScript) public {
+        require(canForward(msg.sender, _evmScript), ERROR_CANNOT_FORWARD);
+        _newVote(_evmScript, "");
+    }
+
+    /**
+    * @notice Tells whether `_sender` can forward actions or not
+    * @dev IForwarder interface conformance
+    * @param _sender Address of the account intending to forward an action
+    * @return True if the given address can create votes, false otherwise
+    */
+    function canForward(address _sender, bytes) public view returns (bool) {
+        // Note that `canPerform()` implicitly does an initialization check itself
+        return canPerform(_sender, CREATE_VOTES_ROLE, arr(_sender));
+    }
+
+    // Disputable app callbacks
+
+    /**
+    * @dev Challenge a vote
+    * @param _voteId Identification number of the vote to be challenged
+    */
+    function _onDisputableActionChallenged(uint256 _voteId, uint256 /* _challengeId */, address /* _challenger */) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_canPause(vote_), ERROR_CANNOT_PAUSE_VOTE);
+
+        vote_.disputableStatus = DisputableStatus.Paused;
+        vote_.pausedAt = getTimestamp64();
+        emit PauseVote(_voteId);
```

Comment:
Could we log the `challengeId` as well?

Having that information here will be really useful:

1. If you're syncing old events, and your vote has been challenged multiple times, the `currentChallengeId` will not reflect that old event's challengeId
2. It allows us to potentially skip a roundtrip call to `getAction()` (which causes 1) and directly call `getChallenge()`